src/theory/arith/nl/nl_model.cpp

```diff
@@ -513,31 +511,27 @@ bool NlModel::solveEqualitySimple(Node eq,
   }
 
   // we are linear, it is simple
-  if (a == d_zero)
+  if (b == d_zero)
```

Comment:
Can you add `Assert(a == d_zero)` then?

Reply:
I deleted `a`.  This was the coefficient used for the old solving of quadratics, which was only partially deleted previously.